machine-config/harvester.vue

```diff
@@ -47,10 +47,10 @@ export default {
 
     try {
       this.credential = await this.$store.dispatch('rancher/find', { type: NORMAN.CLOUD_CREDENTIAL, id: this.credentialId });
-      const clusterId = get(this.credential, `metadata.annotations."${ HCI_ANNOTATIONS.CLUSTER_ID }"`);
+      const clusterId = get(this.credential, 'decodedData.clusterId');
```

Comment:
@yasker can the driver be updated to handle this without a credential?

Reply:
We found this is a common issue that also exists for the other provisioners e.g like VMWare vSphere, for now, the Harvester users can just create their own cloud credentials for that cluster and then modify the cluster accordingly since it is just a kubeconfig file. cc @vincent99 any good suggestions, thanks
![image](https://user-images.githubusercontent.com/4569037/133286932-d2b0790b-62c7-45a8-80d3-69c165c340bb.png)

Reply:
@richard-cox can you take a look here with Vince out to see what can be done?

Reply:
It looks like `harvestercredentialConfig` is available and contains both `clusterId` and `clusterType` (confirmed with creds created with older ember ui and via dashboard using this branch). Can that be used in places to identify id/type?

Reply:
Both the `clusterId` and `clusterType` are stored in the `harvestercredentialConfig ` of the `cloudCredentials` object, but it is a common issue that user B doesn't have permission to access user A's `cloudCredentails` object due to Rancher's RBAC design. 

Currently, Harvester or other provisioners will require the admin users to create their own `cloudCredentials` if they want to edit this cluster properly.

I have filed an issue to track this scenario https://github.com/rancher/rancher/issues/34800, and since it is a common case here so I would suggest us to merge this PR first if there is no other outstanding issue, and we can address this common issue separately via  https://github.com/rancher/rancher/issues/34800.

Reply:
A user granted cluster ownership of someone else's cluster must be able to edit that cluster without access to the original credential or knowledge of the contents of it to recreate their own.  Every other driver works this way.

They make requests through /meta/proxy or other special endpoints that accept a credential *ID*, validate cluster membership, and then generate the request and proxy the response.

Reply:
> They make requests through /meta/proxy or other special endpoints that accept a credential ID, validate cluster membership, and then generate the request and proxy the response.

Is this something that needs to be implemented on the node driver side or it is already built-in support from the Rancher side, would be appreciated if we can take a look of an example, thanks @vincent99

Reply:
`/meta/proxy` is built into Rancher, and has various [methods](https://github.com/rancher/rancher/blob/master/pkg/httpproxy/sign.go#L41-L56) of adding auth info from the secret to the request to be proxied.  This is used in all the other UI drivers to make requests to the cloud provider ([AWS](https://github.com/rancher/dashboard/blob/master/store/aws.js#L19), [DO](https://github.com/rancher/dashboard/blob/master/store/digitalocean.js#L170), etc).

If none of those are appropriate you can add a new signing method.  Or some things like AKS just have their own [special endpoints](https://github.com/rancher/rancher/blob/master/pkg/api/norman/customization/aks/handler.go#L65) to do whatever specific behavior is needed instead of generic proxying.

The important part is an owner of the cluster can use the proxy to make requests for info about that cluster/IaaS provider by saying to use that cloudCredentialId, without direct access to actually get the credential or its contents.

Reply:
I think it makes sense as it will provide a better user experience to the Harvester node driver users, but apparently, this is not a small change and will require a PR submitted to the Rancher repo to support this feature, @yasker I don't think we have enough time to get it complete by the end of this week and in the Rancher 2.6.1, so just prefer to move this to the next release as a feature enhancement, WDUT?